torchgeo/datamodules/etci2021.py

```diff
@@ -153,4 +151,4 @@ def test_dataloader(self) -> DataLoader[Any]:
 
     def plot(self, *args: Any, **kwargs: Any) -> plt.Figure:
         """Run :meth:`torchgeo.datasets.ETCI2021.plot`."""
-        return self.val_dataset.plot(*args, **kwargs)
+        return self.test_dataset.plot(*args, **kwargs)
```

Comment:
I think this is the first bug I've found and fixed thanks to mypy! `val_dataset` doesn't have a `plot` method because it's a subset, you either have to use `val_dataset.dataset.plot` or `test_dataset.plot` (the latter was easier due to type hints). I also found a bug in our ETCI2021 plot function thanks to this.

The reason this bug wasn't found by the unit tests is because we have a try-except that ignores AttributeErrors. This is needed for pytorch-lightning reasons, but unfortunately it also means that missing plot methods were also ignored.

Reply:
@calebrob6 this is one of the ETCI 2021 bugs